Source/Public/G1TFormatConvert.h

```diff
@@ -0,0 +1,4550 @@
+    };
+
+    constexpr static CustomSwizzleItem itemsVary128[] =
+    {
+        {BlockAxis::LinearX, 1}, // 1x1
+        {BlockAxis::LinearY, 1}, // 1x1
+        {BlockAxis::MortonX, 2},
+        {BlockAxis::MortonY, 4},
+        {BlockAxis::MortonX, 4},
+        {BlockAxis::MortonY, 2},
+        {BlockAxis::MortonX, 2},
+        {BlockAxis::MortonY, 2},
+        {BlockAxis::MortonX, 2},
+        {BlockAxis::MortonY, 2},
+        {BlockAxis::MortonX, 2},
+        {BlockAxis::MortonY, 2},
+        {BlockAxis::MortonX, 2},
+    };
+
+    struct
+    {
+        CustomSwizzleItem const* data = nullptr;
+        size_t count = 0;
+    } items;
+
+    if (useBlock)
+    {
+        switch (texelByteSize)
+        {
+        default:
+        case 64:   items = { itemsVary64, countof(itemsVary64) };   break;
+        //case 128:  items = { itemsVary128, countof(itemsVary128) };  break;
+        }
+    }
+    else
+    {
+        switch (texelByteSize)
+        {
+        default:
+        case 8:   items = { items8bpp, countof(items8bpp) };    break;
+        case 16:  items = { items16bpp, countof(items16bpp) };  break;
+        case 32:  items = { items32bpp, countof(items32bpp) };  break;
+        case 64:
+            if (blockPixelSize == 4) // 4x4 block compressed.
+            {
+                items = { items64bppBc, countof(items64bppBc) };
+            }
+            else // Pixels
+            {
+                items = { items64bpp, countof(items64bpp) };
+            }
+            break;
+        case 128:
+            if (blockPixelSize == 4) // 4x4 block compressed.
+            {
+                items = { items128bppBc, countof(items128bppBc) };
+            }
+            else // Pixels
+            {
+                items = { items128bpp, countof(items128bpp) };
+            }
+            break;
+        }
+    }
+        
+    return CustomSwizzle(width, height, x, y, items.data, items.count);
+}
+
+uint32_t Ps2Unswizzle8bpp(uint32_t width, uint32_t height, uint32_t x, uint32_t y)
+{
+    // From http://ps2linux.no-ip.info/playstation2-linux.com/docs/howto/display_docef7c.html?docid=75
+    //
+    // this function works for the following resolutions
+    // Width:       any multiple of 16 smaller than or equal to 4096
+    // Height:      any multiple of 4 smaller than or equal to 4096
+
+    // the texels must be uploaded as a 32bit texture
+    // width_32bit = width_8bit / 2
+    // height_32bit = height_8bit / 2
+    // remember to adjust the mapping coordinates when
+    // using a dimension which is not a power of two
+
+#if 0
+    const uint32_t blockLocation = (y & (~0xF)) * width + (x & (~0xF)) * 2;
+    const uint32_t swapSelector = (((y + 2) >> 2) & 0x1) * 4;
+    const uint32_t posY = (((y & (~3)) >> 1) + (y & 1)) & 0x7;
+    const uint32_t columnLocation = posY * width * 2 + ((x + swapSelector) & 0x7) * 4;
+    const uint32_t byteNumber = ((y >> 1) & 1) + ((x >> 2) & 2); // 0,1,2,3
+#else
```

Comment:
You can delete this. I just expanded the bit patterns below to more easily identify where the bits were going, but the code is functionally identically.